src/Altinn.Platform/Altinn.Platform.Events/Events/Controllers/PushController.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.Common.AccessToken.Configuration;
+using Altinn.Common.PEP.Interfaces;
+using Altinn.Platform.Events.Authorization;
+using Altinn.Platform.Events.Configuration;
+using Altinn.Platform.Events.Models;
+using Altinn.Platform.Events.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Controllers
+{
+    /// <summary>
+    /// Controller responsible for pushing events to subscribers
+    /// </summary>
+    [Route("events/api/v1/push")]
+    [ApiController]
+    public class PushController : ControllerBase
+    {
+        private readonly ISubscriptionService _subscriptionService;
+        private readonly IEventsService _eventsService;
+
+        private readonly AuthorizationHelper _authorizationHelper;
+        private readonly PlatformSettings _platformSettings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PushController"/> class.
+        /// </summary>
+        public PushController(
+        IEventsService eventsService,
+        ISubscriptionService subscriptionService,
+        IPDP pdp,
+        IOptions<PlatformSettings> platformSettings)
+        {
+            _eventsService = eventsService;
+            _subscriptionService = subscriptionService;
+            _authorizationHelper = new AuthorizationHelper(pdp);
+            _platformSettings = platformSettings.Value;
+        }
+
+        /// <summary>
+        /// Alert push controller about a new event
+        /// </summary>
+        /// <returns>Http status</returns>
+        [Authorize]
+        [HttpPost]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Produces("application/json")]
+        public async Task<ActionResult> Post([FromBody] CloudEvent cloudEvent)
```

Comment:
This is a public endpoint. It might not be meant to be used by anyone except our own push mechanism, but I still would like to see complete documentation.

Reply:
Added more documentation